### ISSUE TITLE
xml2json not working with files produced by PrintXML

### ISSUE THREAD

**Issue by beojan** [diana-hep/pyhf#618]
# Description

xml2json isn't working. With a single channel file, an empty skeleton is returned:
```
╭─   beojan   ~/DPhil/hh4b/Resolved/Limits                                                                                                                         SIGINT(2) ↵  54.75M   11:59:30 
╰─ pyhf xml2json meas_scalar_251_xml/Measurement_resolved_4b_2017.xml
0channel [00:00, ?channel/s]
{
    "channels": [],
    "measurements": [],
    "observations": [],
    "version": "1.0.0"
}
```

With the multi channel file, I get an `IsADirectoryError`:
```
IsADirectoryError: [Errno 21] Is a directory: '/home/beojan/DPhil/hh4b/Resolved/Limits/'
```

# Expected Behavior

In each case, JSON would be returned describing the full model.

# Actual Behavior

An empty JSON skeleton is returned, or an `IsADirectoryError` is thrown.

# Steps to Reproduce

1. Write a HistFactory measurement to a ROOT file with `meas.writeToFile`
2. Load this file in a new ROOT session and write the XML: `meas->PrintXML("meas")`
3. Attempt to convert to JSON: `pyhf xml2json meas/meas.xml`

PyHF is installed into an empty virtualenv with Python 3.7.4. The PyHF version is 0.1.2, but the problem also appears after installing from git master.

# Checklist
- [x] Run `git fetch` to get the most up to date version of `master`
- [x] Searched through existing Issues to confirm this is not a duplicate issue
- [x] Filled out the Description, Expected Behavior, Actual Behavior, and Steps to Reproduce sections above or have edited/removed them in a way that fully describes the issue

**Comment by lukasheinrich** [comment]
thanks @beojan for the report. Can you make the XML and ROOT files available somewhere public (for a toy example there should not be any access issues, no need to use a ATLAS internal example) ? We can then debug. Probably  it's some detail that we're not covering for that specific measurement.

**Comment by beojan** [comment]
Attached. I generated dummy data, but the structure is the same. Beware, there's no top level directory in this file, just a ROOT file and the XML file.

[test.tar.gz](https://github.com/diana-hep/pyhf/files/3750452/test.tar.gz)

**Comment by beojan** [comment]
The combination XML is also attached here. 
[Measurement.xml.txt](https://github.com/diana-hep/pyhf/files/3750468/Measurement.xml.txt)

In the combination case, the XML doesn't even load, so it's probably not a problem with the specific measurement.

**Comment by lukasheinrich** [comment]
Hi, @beojan, the reason is that `InputFile` is not specified for the `<Data>` section. Is this intentional? (could be related to #566 )

**Comment by beojan** [comment]
Yes, because the measurement is blinded (I'm just looking for expected limits).

**Comment by lukasheinrich** [comment]
Thanks. Can you (just as a test) try adding a dummy data histogram (with
empty data.. say all zero counts) and see if that works? To be clear you
should always run `xml2json` on the measurement xml, not the individual
channels.

**Comment by beojan** [comment]
I reused the background histogram as the data, and it works. Thanks.

**Comment by kratsg** [comment]
./cc @alexander-held who ran into this issue as well.

**Comment by alexander-held** [comment]
related: #566
